core/offchain/src/lib.rs

```diff
@@ -134,19 +137,18 @@ impl<Client, Storage, Block> OffchainWorkers<
 			futures::future::Either::Right(futures::future::ready(()))
 		}
 	}
-}
 
-/// Spawns a new offchain worker.
-///
-/// We spawn offchain workers for each block in a separate thread,
-/// since they can run for a significant amount of time
-/// in a blocking fashion and we don't want to block the runtime.
-///
-/// Note that we should avoid that if we switch to future-based runtime in the future,
-/// alternatively:
-/// TODO [ToDr] (#1458) we can consider using a thread pool instead.
-fn spawn_worker(f: impl FnOnce() -> () + Send + 'static) {
-	std::thread::spawn(f);
+	/// Spawns a new offchain worker.
+	///
+	/// We spawn offchain workers for each block in a separate thread,
+	/// since they can run for a significant amount of time
+	/// in a blocking fashion and we don't want to block the runtime.
+	///
+	/// Note that we should avoid that if we switch to future-based runtime in the future,
+	/// alternatively:
+	fn spawn_worker(&self, f: impl FnOnce() -> () + Send + 'static) {
+		self.thread_pool.lock().unwrap().execute(f);
```

Comment:
When you switch to `parking_lot::Mutex`, this will be just:
```suggestion
		self.thread_pool.lock().execute(f);
```